docs/DeveloperGuide.md

```diff
@@ -141,20 +145,60 @@ It also constructs `CliMessages` to display messages to the user.
     - Cons: It is unoptimized in terms of complexity, which requires more work for scaling of the application.
 - **Alternative 2:** Stores as a Hash Table with the key as the index and value as `Task`
     - Pros: It has a better time complexity and reduce the work in scaling stage since this data structure is more optimized (O(1) can be achieved).
-    - Cons: It takes more resources to implement. Furthermore, 
+    - Cons: It takes more resources to implement. Furthermore,
 
 ### 4.2.2. List feature
 
 #### 4.2.2.1. Implementation
 
 The following sequence diagram shows how the list operation works:
 
-<img src="https://github.com/longngng/tp/blob/longngng-DG/docs/diagrams/class/jpeg/SequenceDiagram_List.jpg">
+![Sequence Diagram for List command](/diagrams/class/jpeg/SequenceDiagram_List.jpg)
 
-#### 4.2.2.2. Design Considerations
 
 ### 4.3. Notebook Mode
 
+
+### 4.4 [Proposed] Find duplicate feature
+
+#### 4.4.1 Proposed implementation
+
+The proposed find duplicate function is facilitated by a method in the classes `Task List`, `Notebook Shelf`, `Notebook` and `Section`.
+
+Given below is an example usage scenario and how the find duplicates function behaves.
+
+Step 1. The user launches the application for the first time. CliUserInterface#executeCommand is called when the user adds a task into the task list.
+
+Step 2. The user types `add /tTask /by19-10-2020 1900`. The `add` command is passed through `InputParser#getCommandFromInput`, which then calls `AddCommandTimetableMode#execute()`.
+
+Step 3. `execute()` is called, which then calls `InputParser#parseTaskTitle`, which first extracts the `title` from the user's input.
+
+Step 4. The `title` is then passed to the `findDuplicate` method in `TaskList`.
+
+Step 5. The `findDuplicate` method returns false, since it is the first task titled `Task` to be added into the `TaskList`. Conversely, the `findDuplicate` method returns true when a task with the same `title` already exists in the `TaskList`.
+
+Step 6. `InputParser#parseDeadline` is then called, which returns the `deadline` to `AddCommandTimetableMode#execute()`.
+
+Step 7. `TaskList#addTask` is then called and a new `Task`, with `title` and `deadline`, is initialised.
+
+Step 8. To signal that the user has successfully added a task, a message is printed with `CliMessages#printAddedTaskMessage`.
+
+
+The sequence diagram below shows how the find duplicate command works:
+
+![Sequence diagram for finding duplicates](/diagrams/class/jpeg/duplicates_francene.jpg)
```

Comment:
Image link appears to be dead, perhaps merge first and then fix to get the final URL